domain/secretbackend/state/state.go

```diff
@@ -447,49 +457,84 @@ func (s *State) GetCloudCredential(ctx context.Context, modelUUID coremodel.UUID
 	if err != nil {
 		return cld, cred, errors.Trace(err)
 	}
-	modelStmt, err := s.Prepare(`
-SELECT &ModelCloudCredentialRow.*
-FROM v_model
-WHERE uuid = $M.uuid`, sqlair.M{}, ModelCloudCredentialRow{})
+	stmt, err := s.Prepare(`
+SELECT (
```

Comment:
We're selecting too much here. Ultimately, this is used to make a k8s provider config, so we just need enough to satisfy
```
func cloudSpecToBackendConfig(spec cloudspec.CloudSpec) (*provider.BackendConfig, error) {
	cred, err := json.Marshal(spec.Credential)
	if err != nil {
		return nil, errors.Trace(err)
	}
	return &provider.BackendConfig{
		BackendType: BackendType,
		Config: map[string]interface{}{
			"endpoint":            spec.Endpoint,
			"ca-certs":            spec.CACertificates,
			"is-controller-cloud": spec.IsControllerCloud,
			"credential":          string(cred),
		},
	}, nil
}
```

All the storage endpoint and revoked and invalid and owner etc things not needed should not be selected.